Update demos to demonstrate breaking edge cases

alibosworth opened the pull request:
This helps diagnose and fix edge cases that are currently broken

alibosworth committed:
Update the sortable demo to include header and footer elements

These elements are within the root sortable element but are not
sortable themselves. Sortable items should be draggable within
eachother but not be inserted before the header or after the
footer.

alibosworth committed:
Add demo with nested sortable elements

This demo shows an example of using two sortable levels:
a top-level section sortable and within each section there
are nested sortable items.  The two sortables should
operated independantly,

Rosey commented:
I know you wanted to merge this ahead of the other so we can easily see the difference, but I feel like the commit of the demo doesn't need to talk about how there's a bug, since there's a bug and we're fixing that and these demos will continue to be useful even after the bug is fixed. Seems more like we can just say something about how it's a demo of how sortable works when there are other elements nested under the parent, and a demo of how sortable works when there's nested sortable elements?

alibosworth committed:
Add child elements to sortable demo

Even if the child elements are the source of the drag (and only their
parents match the item selector) the correct items should drag

alibosworth commented:
👍  I'll reword

alibosworth commented:
reworded
